app/controllers/work_packages_controller.rb

```diff
@@ -83,12 +83,16 @@ def index
   protected
 
   def export_list(mime_type)
-    export_storage = WorkPackages::Exports::ScheduleService
+    job_id = WorkPackages::Exports::ScheduleService
                      .new(user: current_user)
                      .call(query: @query, mime_type: mime_type, params: params)
                      .result
 
-    redirect_to work_packages_export_path(export_storage.id)
+    if request.headers['Accept']&.include?('application/json')
```

Comment:
That's because we override the format with the route constraint (e.g., `.xls` will result in excel mime type). I haven't found a way to differentiate between the request format and the accept header.